server/src/main/java/com/google/musicanalysis/site/YouTubeCallbackServlet.java

```diff
@@ -27,7 +39,14 @@ protected String getTokenUri() {
 
   @Override
   protected String getRedirectUri() {
-    return System.getenv().get("YOUTUBE_CALLBACK_URI");
+    // URI that user should be redirected to after logging in
+    try {
+      var domainUri = URI.create(System.getenv().get("DOMAIN"));
+      return domainUri.resolve("/api/oauth/callback/youtube").toString();
```

Comment:
Also I had some redirect_uri error when trying to do the oauth, so I had to add `/api/oauth/callback/youtube` and `api/oauth/callback/spotify` as possible redirect uris to the cloud project instead of only `api/oauth/callback`

Reply:
yeah, I tried to add it myself but I couldn't find where it was

Reply:
okay I changed it but didn't know if that was needed